lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py

```diff
@@ -125,21 +172,46 @@ def tool(self, pbench_run):
             tool.write_text("--interval=30")
             tool = p / "mpstat"
             tool.write_text("--interval=300")
-        return
+            if group == "default":
+                tool = p / "sar"
+                tool.write_text("--interval=10")
+            else:
+                tool = p / "perf"
+                tool.write_text("--record-opts='-a --freq=100'")
+
+    @pytest.fixture
+    def tools_on_multiple_hosts(self, pbench_run):
+        for group in ["default", "test"]:
+            for host in ["th1.example.com", "th2.example.com"]:
+                p = pbench_run / f"tools-v1-{group}" / host
+                p.mkdir(parents=True)
+                tool = p / "iostat"
+                tool.write_text("--interval=30")
+                tool = p / "mpstat"
+                tool.write_text("--interval=300")
+                if host == "th1.example.com":
+                    tool = p / "sar"
+                    tool.write_text("--interval=10")
+                else:
+                    tool = p / "perf"
+                    tool.write_text("--record-opts='-a --freq=100'")
```

Comment:
Do you really want each of the two groups configured identically?

Given that you have two hosts in each of two groups, I recommend generating four distinct configurations (perhaps different hosts in the different groups and different tools for the different hosts; you can put the group name in the host name to make them easy to generate and easier to correlate).

Reply:
Too complicated for Friday afternoon: I'm going to punt on this until the `list.py` refactoring.

Reply:
I think all you have to do is change two lines.  Change line 187 to something like this:
```
            for host in [f"{group}1.example.com", f"{group}2.example.com"]:
```
so that the `default` group will have hosts `default1.example.com` and `default2.example.com` while the `test` group will have hosts`test1.example.com` and `test2.example.com`; and then change line 194 to match:
```
                if host == f"{group}1.example.com":
```
or you can change it to check specifically for `default1.example.com` and add another case which checks for `test1.example.com`.